### PR TITLE
Types: import more *RegistrationOptions types from LSP spec

### PR DESCRIPTION
**Pull request by razzmatazz**
This PR imports some of *RegistrationOptions types from LSP spec

### CODE REVIEW

**Comment by razzmatazz** [comment]
Ideally https://github.com/ionide/LanguageServerProtocol/pull/49 would do this for us, but for now..

**Comment by razzmatazz** [comment]
CC @baronfel

**Comment by TheAngryByrd** [comment]
> Ideally #49 would do this for us, but for now..

Yeah I need to get on that. one day...

File: src/Types.fs [hunk 1]
```diff
@@ -1512,6 +1521,13 @@ type SignatureHelpOptions =
     RetriggerCharacters: char[] option
   }
 
+type SignatureHelpRegistrationOptions =
```
Comment (TheAngryByrd) [inline src/Types.fs:1524]: Same here about [workDoneProgress](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#workDoneProgressOptions)

File: src/Types.fs [hunk 2]
```diff
@@ -1500,6 +1500,15 @@ type CompletionOptions =
     CompletionItem: CompletionItemOptions option
   }
 
+type CompletionRegistrationOptions =
```
[the remark this answers is not in the document]
Comment (TheAngryByrd) [reply src/Types.fs:1503]: There's 28 types that seem to extend `WorkDoneProgressOptions` and none of our current types care about it.

I'm ok without having it for this PR.

**Review by TheAngryByrd — APPROVED** [review]

**Pull request merged**